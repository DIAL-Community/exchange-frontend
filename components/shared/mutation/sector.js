import { gql } from '@apollo/client'

export const DELETE_SECTOR = gql`
  mutation DeleteSector($id: ID!) {
    deleteSector(id: $id) {
      sector {
       id
       slug
       name
      }
      errors
    }
  }
`

export const CREATE_SECTOR = gql`
  mutation CreateSector(
    $name: String!
    $slug: String!
    $isDisplayable: Boolean!
    $locale: String!
  ) {
    createSector(
      name: $name
      slug: $slug
      isDisplayable: $isDisplayable
      locale: $locale
    ) {
      sector {
        id
        name
        slug
        locale
        isDisplayable
      }
      errors
    }
  }
`
