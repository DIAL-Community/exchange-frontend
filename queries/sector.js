import { gql } from '@apollo/client'

export const SECTOR_SEARCH_QUERY = gql`
  query Sectors($search: String!, $locale: String) {
    sectors(search: $search, locale: $locale) {
      id
      name
      slug
    }
  }
`
