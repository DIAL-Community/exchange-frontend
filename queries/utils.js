const MIN_SEARCH_VALUE_CHARACTERS = 2

export const fetchSelectOptions = async (client, input, query, fetchedDataCallback, locale = null) => {
  const variables = {
    search: input
  }

  if (input && input.trim().length < MIN_SEARCH_VALUE_CHARACTERS) {
    return []
  } else if (locale) {
    variables.locale = locale
  }

  const response = await client.query({ query, variables })

  return response.data ? fetchedDataCallback(response.data) : []
}

export const fetchSelectOptionsWithMature = async (client, input, query, fetchedDataCallback, locale = null) => {
  const variables = {
    search: input,
    mature: true
  }

  if (input && input.trim().length < MIN_SEARCH_VALUE_CHARACTERS) {
    return []
  } else if (locale) {
    variables.locale = locale
  }

  const response = await client.query({ query, variables })

  return response.data ? fetchedDataCallback(response.data) : []
}
