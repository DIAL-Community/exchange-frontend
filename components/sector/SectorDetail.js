import { useRef } from 'react'
import { useApolloClient, useQuery } from '@apollo/client'
import { GRAPH_QUERY_CONTEXT } from '../../lib/apolloClient'
import Breadcrumb from '../shared/Breadcrumb'
import { handleLoadingQuery, handleMissingData, handleQueryError } from '../shared/GraphQueryHandler'
import { SECTOR_DETAIL_QUERY, SECTOR_POLICY_QUERY } from '../shared/query/sector'
import { fetchOperationPolicies } from '../utils/policy'
import SectorDetailLeft from './SectorDetailLeft'
import SectorDetailRight from './SectorDetailRight'

const SectorDetail = ({ slug }) => {
  const scrollRef = useRef(null)
  const client = useApolloClient()

  const policies = fetchOperationPolicies(
    client,
    SECTOR_POLICY_QUERY,
    ['editing', 'deleting']
  )

  const editingAllowed = policies['editing']
  const deletingAllowed = policies['deleting']

  const { loading, error, data } = useQuery(SECTOR_DETAIL_QUERY, {
    variables: { slug },
    context: {
      headers: {
        ...GRAPH_QUERY_CONTEXT.VIEWING
      }
    }
  })

  if (loading) {
    return handleLoadingQuery()
  } else if (error) {
    return handleQueryError(error)
  } else if (!data?.sector) {
    return handleMissingData()
  }

  const { sector } = data

  const slugNameMapping = (() => {
    const map = {}
    map[sector.slug] = sector.name

    return map
  })()

  return (
    <div className='lg:px-8 xl:px-56 flex flex-col'>
      <div className='px-4 lg:px-6 py-4 bg-dial-violet text-dial-stratos ribbon-detail z-40'>
        <Breadcrumb slugNameMapping={slugNameMapping}/>
      </div>
      <div className='flex flex-col lg:flex-row gap-x-8'>
        <div className='lg:basis-1/3'>
          <SectorDetailLeft scrollRef={scrollRef} sector={sector} />
        </div>
        <div className='lg:basis-2/3'>
          <SectorDetailRight
            ref={scrollRef}
            sector={sector}
            editingAllowed={editingAllowed}
            deletingAllowed={deletingAllowed}
          />
        </div>
      </div>
    </div>
  )
}

export default SectorDetail
