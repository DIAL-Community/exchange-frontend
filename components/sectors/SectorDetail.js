import { useCallback } from 'react'
import { useIntl } from 'react-intl'
import { useQuery } from '@apollo/client'
import { useRouter } from 'next/router'
import Dialog from '../shared/Dialog'
import Checkbox from '../shared/Checkbox'
import { SECTOR_SEARCH_QUERY } from '../../queries/sector'
import { getLanguageOptions } from '../../lib/utilities'

const labelTextBlue = 'text-dial-blue'
const labelTextGray = 'text-button-gray'

const SectorDetail = ({ sector, isOpen, onClose }) => {
  const { locale } = useRouter()
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  const { data } = useQuery(SECTOR_SEARCH_QUERY, {
    variables: { search: '', locale }
  })

  const parentSector = data?.sectors?.find(({ id }) => id === sector?.parentSectorId?.toString())
  const sectorLocale = getLanguageOptions(format).find(({ value } ) => value === sector?.locale)

  return (
    <Dialog
      closeButton
      isOpen={isOpen}
      onClose={onClose}
    >
      <div className='flex flex-col gap-3' data-testid='sector-detail'>
        <div className={`${labelTextBlue} text-2xl font-bold pb-4`}>
          <p data-testid='sector-name'>{`${format('sector.label')}: ${sector.name}`}</p>
        </div>
        <div className='flex flex-col gap-3'>
          {sector?.parentSectorId && (
            <div className='text-xl'>
              <p className={labelTextBlue}>{`${format('sector.parent-sector.label')}:`}</p>
              <p className={labelTextGray} data-testid='parent-sector'>{parentSector?.name}</p>
            </div>
          )}
          <div className='text-xl'>
            <p className={labelTextBlue}>{`${format('locale.label')}:`}</p>
            <p className={labelTextGray} data-testid='sector-locale'>{sectorLocale.label}</p>
          </div>
          <div className={'flex flex-row text-xl gap-x-2'}>
            <Checkbox value={sector.isDisplayable} disabled/>
            <p className={labelTextBlue}>{format('sector.is-displayable.label')}</p>
          </div>
        </div>
      </div>  
    </Dialog>
  )
}

export default SectorDetail
