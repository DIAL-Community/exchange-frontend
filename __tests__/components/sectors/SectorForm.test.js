import { act } from 'react-dom/test-utils'
import { fireEvent, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { mockObserverImplementation, waitForAllEffects, render } from '../../test-utils'
import CustomMockedProvider, { generateMockApolloData } from '../../utils/CustomMockedProvider'
import SectorForm from '../../../components/sectors/SectorForm'
import { SECTOR_SEARCH_QUERY } from '../../../queries/sector'
import { mockNextAuthUseSession, mockNextUseRouter, statuses } from '../../utils/nextMockImplementation'
import { sectors, sectorWithoutParentSector, sectorWithParentSector } from './data/SectorDetail'

mockNextUseRouter()
describe('Unit tests for the SectorForm component.', () => {
  const DIALOG_FORM_TEST_ID = 'dialog'
  const SECTOR_NAME_TEST_ID = 'sector-name'
  const SECTOR_LOCALE_TEST_ID = 'sector-locale'
  const REQUIRED_FIELD_MESSAGE = 'This field is required'
  const mockedSectors = generateMockApolloData(SECTOR_SEARCH_QUERY, { search: '', locale: 'en' }, null, sectors)
  const mockOnClose = jest.fn()

  beforeAll(() => {
    mockNextAuthUseSession(statuses.AUTHENTICATED, { canEdit: true })
    window.IntersectionObserver = mockObserverImplementation()
  })

  describe('Should match snapshot -', () => {
    test('create.', async () => {
      const { getByTestId } = render(
        <CustomMockedProvider mocks={[mockedSectors]}>
          <SectorForm
            isOpen={true}
            onClose={mockOnClose}
          />
        </CustomMockedProvider>
      )
      await waitForAllEffects(1000)
      expect(getByTestId(DIALOG_FORM_TEST_ID)).toMatchSnapshot()
    })

    test('edit for sector which has parent sector.', async () => {
      const { getByTestId } = render(
        <CustomMockedProvider mocks={[mockedSectors]}>
          <SectorForm
            isOpen={true}
            onClose={mockOnClose}
            sector={sectorWithParentSector}
          />
        </CustomMockedProvider>
      )
      await waitForAllEffects(100)
      expect(getByTestId(DIALOG_FORM_TEST_ID)).toMatchSnapshot()
    })

    test('edit for which does not have parent sector.', async () => {
      const { getByTestId } = render(
        <CustomMockedProvider mocks={[mockedSectors]}>
          <SectorForm
            isOpen={true}
            onClose={mockOnClose}
            sector={sectorWithoutParentSector}
          />
        </CustomMockedProvider>
      )
      await waitForAllEffects(100)
      expect(getByTestId(DIALOG_FORM_TEST_ID)).toMatchSnapshot()
    })
  })

  describe('For mandatory field -', () => {
    test('should show validation errors.', async () => {
      const { getByTestId, getByText } = render(
        <CustomMockedProvider mocks={[mockedSectors]}>
          <SectorForm
            isOpen={true}
            onClose={mockOnClose}
          />
        </CustomMockedProvider>
      )
      await waitForAllEffects()
      await act(async () => fireEvent.click(getByText('Submit')))
      expect(getByTestId(SECTOR_NAME_TEST_ID)).toHaveTextContent(REQUIRED_FIELD_MESSAGE)
      expect(getByTestId(SECTOR_LOCALE_TEST_ID)).not.toHaveTextContent(REQUIRED_FIELD_MESSAGE)
    })

    test('should show validation errors and hide one of them on input value change.', async () => {
      const user = userEvent.setup()
      const { getByTestId, getByText } = render(
        <CustomMockedProvider mocks={[mockedSectors]}>
          <SectorForm
            isOpen={true}
            onClose={mockOnClose}
          />
        </CustomMockedProvider>
      )
      await waitForAllEffects()
      await act(async () => fireEvent.click(getByText('Submit')))
      expect(getByTestId(SECTOR_NAME_TEST_ID)).toHaveTextContent(REQUIRED_FIELD_MESSAGE)
      expect(getByTestId(SECTOR_LOCALE_TEST_ID)).not.toHaveTextContent(REQUIRED_FIELD_MESSAGE)

      await user.type(screen.getByLabelText(/Name/), 'test sector name')
      expect(getByTestId(SECTOR_NAME_TEST_ID)).not.toHaveTextContent(REQUIRED_FIELD_MESSAGE)
      expect(getByTestId(SECTOR_LOCALE_TEST_ID)).not.toHaveTextContent(REQUIRED_FIELD_MESSAGE)
    })
  })
})
