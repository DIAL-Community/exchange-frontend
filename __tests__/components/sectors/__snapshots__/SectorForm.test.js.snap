// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for the SectorForm component. Should match snapshot - create. 1`] = `
<div
  aria-describedby="headlessui-description-5"
  aria-modal="true"
  class="fixed z-100"
  data-testid="dialog"
  id="headlessui-dialog-1"
  role="dialog"
>
  <div
    class="fixed inset-0 bg-dial-gray opacity-80"
  />
  <div
    class="fixed inset-0 flex items-center justify-center"
  >
    <div
      class="max-h-screen w-screen md:w-3/5 lg:w-3/5 xl:w-1/2 p-8 transform bg-white shadow-xl rounded-2xl flex flex-col opacity-100 scale-100"
      id="headlessui-dialog-panel-4"
    >
      <div
        class="bg-edit overflow-auto py-4 px-8 flex flex-col gap-3"
        id="headlessui-description-5"
      >
        <div
          class="flex justify-start text-xl"
          data-testid="dialog-body"
        >
          <div
            class="w-full"
          >
            <form
              id="sector-form"
            >
              <div
                class="pb-12 mb-4 flex flex-col gap-3"
              >
                <div
                  class="text-2xl font-bold text-dial-blue pb-4"
                >
                  Create New Sector
                </div>
                <div
                  class="flex flex-col gap-y-2 mb-2"
                  data-testid="sector-name"
                >
                  <label
                    class="text-xl text-dial-blue required-field"
                    for="name"
                  >
                    Name
                  </label>
                  <input
                    class="w-full"
                    id="name"
                    name="name"
                    placeholder="Name"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="flex flex-col gap-y-2 mb-2"
                  data-testid="sector-locale"
                >
                  <label
                    class="text-xl text-dial-blue required-field"
                    for="locale"
                  >
                    Locale
                  </label>
                  <div
                    class=" css-l45kjq-container"
                  >
                    <span
                      class="css-1f43avz-a11yText-A11yText"
                      id="react-select-2-live-region"
                    />
                    <span
                      aria-atomic="false"
                      aria-live="polite"
                      aria-relevant="additions text"
                      class="css-1f43avz-a11yText-A11yText"
                    />
                    <div
                      class="react-select__control css-19jej77-control"
                    >
                      <div
                        class="react-select__value-container react-select__value-container--has-value css-3tzzl5-ValueContainer"
                      >
                        <div
                          class="react-select__single-value css-qc6sy-singleValue"
                        >
                          English (en)
                        </div>
                        <div
                          class="react-select__input-container css-6j8wv5-Input"
                          data-value=""
                        >
                          <input
                            aria-autocomplete="list"
                            aria-expanded="false"
                            aria-haspopup="true"
                            autocapitalize="none"
                            autocomplete="off"
                            autocorrect="off"
                            class="react-select__input"
                            id="react-select-2-input"
                            role="combobox"
                            spellcheck="false"
                            style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                            tabindex="0"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                      <div
                        class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                      >
                        <span
                          class="react-select__indicator-separator css-1979yuk-indicatorSeparator"
                        />
                        <div
                          aria-hidden="true"
                          class="react-select__indicator react-select__dropdown-indicator css-1hmmjq6-indicatorContainer"
                        >
                          <svg
                            class="text-2xl"
                            fill="none"
                            height="1em"
                            stroke="currentColor"
                            stroke-width="0"
                            viewBox="0 0 24 24"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="2"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                    <input
                      name="locale"
                      type="hidden"
                      value="en"
                    />
                  </div>
                </div>
                <div
                  class="flex flex-col gap-y-2 mb-2"
                  data-testid="sector-parent"
                >
                  <label
                    class="text-xl text-dial-blue"
                    for="parent"
                  >
                    Parent Sector
                  </label>
                  <div
                    class=" css-l45kjq-container"
                  >
                    <span
                      class="css-1f43avz-a11yText-A11yText"
                      id="react-select-3-live-region"
                    />
                    <span
                      aria-atomic="false"
                      aria-live="polite"
                      aria-relevant="additions text"
                      class="css-1f43avz-a11yText-A11yText"
                    />
                    <div
                      class="react-select__control css-19jej77-control"
                    >
                      <div
                        class="react-select__value-container css-3tzzl5-ValueContainer"
                      >
                        <div
                          class="react-select__placeholder css-19jynah-placeholder"
                          id="react-select-3-placeholder"
                        >
                          Parent Sector
                        </div>
                        <div
                          class="react-select__input-container css-6j8wv5-Input"
                          data-value=""
                        >
                          <input
                            aria-autocomplete="list"
                            aria-describedby="react-select-3-placeholder"
                            aria-expanded="false"
                            aria-haspopup="true"
                            autocapitalize="none"
                            autocomplete="off"
                            autocorrect="off"
                            class="react-select__input"
                            id="react-select-3-input"
                            role="combobox"
                            spellcheck="false"
                            style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                            tabindex="0"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                      <div
                        class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                      >
                        <span
                          class="react-select__indicator-separator css-1979yuk-indicatorSeparator"
                        />
                        <div
                          aria-hidden="true"
                          class="react-select__indicator react-select__dropdown-indicator css-1hmmjq6-indicatorContainer"
                        >
                          <svg
                            class="text-2xl"
                            fill="none"
                            height="1em"
                            stroke="currentColor"
                            stroke-width="0"
                            viewBox="0 0 24 24"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="2"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                    <input
                      name="parentSector"
                      type="hidden"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="flex flex-col gap-y-2 mb-2"
                  data-testid="sector-displayable"
                >
                  <label
                    class="flex gap-x-2 mb-2 items-center self-start form-field-label"
                    for="sector-displayable"
                  >
                    <input
                      class="h-8 w-8"
                      data-testid="checkbox"
                      name="isDisplayable"
                      type="checkbox"
                    />
                    Is displayable
                  </label>
                </div>
              </div>
            </form>
          </div>
        </div>
        <div
          class="flex justify-start gap-3 text-xl"
        >
          <button
            class="submit-button"
            data-testid="submit-button"
            form="sector-form"
            type="submit"
          >
            Submit
          </button>
          <button
            class="cancel-button"
            data-testid="cancel-button"
            type="button"
          >
            Cancel
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit tests for the SectorForm component. Should match snapshot - edit for sector which has parent sector. 1`] = `
<div
  aria-describedby="headlessui-description-10"
  aria-modal="true"
  class="fixed z-100"
  data-testid="dialog"
  id="headlessui-dialog-7"
  role="dialog"
>
  <div
    class="fixed inset-0 bg-dial-gray opacity-80"
  />
  <div
    class="fixed inset-0 flex items-center justify-center"
  >
    <div
      class="max-h-screen w-screen md:w-3/5 lg:w-3/5 xl:w-1/2 p-8 transform bg-white shadow-xl rounded-2xl flex flex-col opacity-100 scale-100"
      id="headlessui-dialog-panel-9"
    >
      <div
        class="bg-edit overflow-auto py-4 px-8 flex flex-col gap-3"
        id="headlessui-description-10"
      >
        <div
          class="flex justify-start text-xl"
          data-testid="dialog-body"
        >
          <div
            class="w-full"
          >
            <form
              id="sector-form"
            >
              <div
                class="pb-12 mb-4 flex flex-col gap-3"
              >
                <div
                  class="text-2xl font-bold text-dial-blue pb-4"
                >
                  Edit Example Sector
                </div>
                <div
                  class="flex flex-col gap-y-2 mb-2"
                  data-testid="sector-name"
                >
                  <label
                    class="text-xl text-dial-blue required-field"
                    for="name"
                  >
                    Name
                  </label>
                  <input
                    class="w-full"
                    id="name"
                    name="name"
                    placeholder="Name"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="flex flex-col gap-y-2 mb-2"
                  data-testid="sector-locale"
                >
                  <label
                    class="text-xl text-dial-blue required-field"
                    for="locale"
                  >
                    Locale
                  </label>
                  <div
                    class=" css-l45kjq-container"
                  >
                    <span
                      class="css-1f43avz-a11yText-A11yText"
                      id="react-select-4-live-region"
                    />
                    <span
                      aria-atomic="false"
                      aria-live="polite"
                      aria-relevant="additions text"
                      class="css-1f43avz-a11yText-A11yText"
                    />
                    <div
                      class="react-select__control css-19jej77-control"
                    >
                      <div
                        class="react-select__value-container react-select__value-container--has-value css-3tzzl5-ValueContainer"
                      >
                        <div
                          class="react-select__single-value css-qc6sy-singleValue"
                        >
                          English (en)
                        </div>
                        <div
                          class="react-select__input-container css-6j8wv5-Input"
                          data-value=""
                        >
                          <input
                            aria-autocomplete="list"
                            aria-expanded="false"
                            aria-haspopup="true"
                            autocapitalize="none"
                            autocomplete="off"
                            autocorrect="off"
                            class="react-select__input"
                            id="react-select-4-input"
                            role="combobox"
                            spellcheck="false"
                            style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                            tabindex="0"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                      <div
                        class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                      >
                        <span
                          class="react-select__indicator-separator css-1979yuk-indicatorSeparator"
                        />
                        <div
                          aria-hidden="true"
                          class="react-select__indicator react-select__dropdown-indicator css-1hmmjq6-indicatorContainer"
                        >
                          <svg
                            class="text-2xl"
                            fill="none"
                            height="1em"
                            stroke="currentColor"
                            stroke-width="0"
                            viewBox="0 0 24 24"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="2"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                    <input
                      name="locale"
                      type="hidden"
                      value="en"
                    />
                  </div>
                </div>
                <div
                  class="flex flex-col gap-y-2 mb-2"
                  data-testid="sector-parent"
                >
                  <label
                    class="text-xl text-dial-blue"
                    for="parent"
                  >
                    Parent Sector
                  </label>
                  <div
                    class=" css-l45kjq-container"
                  >
                    <span
                      class="css-1f43avz-a11yText-A11yText"
                      id="react-select-5-live-region"
                    />
                    <span
                      aria-atomic="false"
                      aria-live="polite"
                      aria-relevant="additions text"
                      class="css-1f43avz-a11yText-A11yText"
                    />
                    <div
                      class="react-select__control css-19jej77-control"
                    >
                      <div
                        class="react-select__value-container css-3tzzl5-ValueContainer"
                      >
                        <div
                          class="react-select__placeholder css-19jynah-placeholder"
                          id="react-select-5-placeholder"
                        >
                          Parent Sector
                        </div>
                        <div
                          class="react-select__input-container css-6j8wv5-Input"
                          data-value=""
                        >
                          <input
                            aria-autocomplete="list"
                            aria-describedby="react-select-5-placeholder"
                            aria-expanded="false"
                            aria-haspopup="true"
                            autocapitalize="none"
                            autocomplete="off"
                            autocorrect="off"
                            class="react-select__input"
                            id="react-select-5-input"
                            role="combobox"
                            spellcheck="false"
                            style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                            tabindex="0"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                      <div
                        class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                      >
                        <span
                          class="react-select__indicator-separator css-1979yuk-indicatorSeparator"
                        />
                        <div
                          aria-hidden="true"
                          class="react-select__indicator react-select__dropdown-indicator css-1hmmjq6-indicatorContainer"
                        >
                          <svg
                            class="text-2xl"
                            fill="none"
                            height="1em"
                            stroke="currentColor"
                            stroke-width="0"
                            viewBox="0 0 24 24"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="2"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                    <input
                      name="parentSector"
                      type="hidden"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="flex flex-col gap-y-2 mb-2"
                  data-testid="sector-displayable"
                >
                  <label
                    class="flex gap-x-2 mb-2 items-center self-start form-field-label"
                    for="sector-displayable"
                  >
                    <input
                      class="h-8 w-8"
                      data-testid="checkbox"
                      name="isDisplayable"
                      type="checkbox"
                    />
                    Is displayable
                  </label>
                </div>
              </div>
            </form>
          </div>
        </div>
        <div
          class="flex justify-start gap-3 text-xl"
        >
          <button
            class="submit-button"
            data-testid="submit-button"
            form="sector-form"
            type="submit"
          >
            Submit
          </button>
          <button
            class="cancel-button"
            data-testid="cancel-button"
            type="button"
          >
            Cancel
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit tests for the SectorForm component. Should match snapshot - edit for which does not have parent sector. 1`] = `
<div
  aria-describedby="headlessui-description-15"
  aria-modal="true"
  class="fixed z-100"
  data-testid="dialog"
  id="headlessui-dialog-12"
  role="dialog"
>
  <div
    class="fixed inset-0 bg-dial-gray opacity-80"
  />
  <div
    class="fixed inset-0 flex items-center justify-center"
  >
    <div
      class="max-h-screen w-screen md:w-3/5 lg:w-3/5 xl:w-1/2 p-8 transform bg-white shadow-xl rounded-2xl flex flex-col opacity-100 scale-100"
      id="headlessui-dialog-panel-14"
    >
      <div
        class="bg-edit overflow-auto py-4 px-8 flex flex-col gap-3"
        id="headlessui-description-15"
      >
        <div
          class="flex justify-start text-xl"
          data-testid="dialog-body"
        >
          <div
            class="w-full"
          >
            <form
              id="sector-form"
            >
              <div
                class="pb-12 mb-4 flex flex-col gap-3"
              >
                <div
                  class="text-2xl font-bold text-dial-blue pb-4"
                >
                  Edit Example Sector
                </div>
                <div
                  class="flex flex-col gap-y-2 mb-2"
                  data-testid="sector-name"
                >
                  <label
                    class="text-xl text-dial-blue required-field"
                    for="name"
                  >
                    Name
                  </label>
                  <input
                    class="w-full"
                    id="name"
                    name="name"
                    placeholder="Name"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="flex flex-col gap-y-2 mb-2"
                  data-testid="sector-locale"
                >
                  <label
                    class="text-xl text-dial-blue required-field"
                    for="locale"
                  >
                    Locale
                  </label>
                  <div
                    class=" css-l45kjq-container"
                  >
                    <span
                      class="css-1f43avz-a11yText-A11yText"
                      id="react-select-6-live-region"
                    />
                    <span
                      aria-atomic="false"
                      aria-live="polite"
                      aria-relevant="additions text"
                      class="css-1f43avz-a11yText-A11yText"
                    />
                    <div
                      class="react-select__control css-19jej77-control"
                    >
                      <div
                        class="react-select__value-container react-select__value-container--has-value css-3tzzl5-ValueContainer"
                      >
                        <div
                          class="react-select__single-value css-qc6sy-singleValue"
                        >
                          English (en)
                        </div>
                        <div
                          class="react-select__input-container css-6j8wv5-Input"
                          data-value=""
                        >
                          <input
                            aria-autocomplete="list"
                            aria-expanded="false"
                            aria-haspopup="true"
                            autocapitalize="none"
                            autocomplete="off"
                            autocorrect="off"
                            class="react-select__input"
                            id="react-select-6-input"
                            role="combobox"
                            spellcheck="false"
                            style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                            tabindex="0"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                      <div
                        class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                      >
                        <span
                          class="react-select__indicator-separator css-1979yuk-indicatorSeparator"
                        />
                        <div
                          aria-hidden="true"
                          class="react-select__indicator react-select__dropdown-indicator css-1hmmjq6-indicatorContainer"
                        >
                          <svg
                            class="text-2xl"
                            fill="none"
                            height="1em"
                            stroke="currentColor"
                            stroke-width="0"
                            viewBox="0 0 24 24"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="2"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                    <input
                      name="locale"
                      type="hidden"
                      value="en"
                    />
                  </div>
                </div>
                <div
                  class="flex flex-col gap-y-2 mb-2"
                  data-testid="sector-parent"
                >
                  <label
                    class="text-xl text-dial-blue"
                    for="parent"
                  >
                    Parent Sector
                  </label>
                  <div
                    class=" css-l45kjq-container"
                  >
                    <span
                      class="css-1f43avz-a11yText-A11yText"
                      id="react-select-7-live-region"
                    />
                    <span
                      aria-atomic="false"
                      aria-live="polite"
                      aria-relevant="additions text"
                      class="css-1f43avz-a11yText-A11yText"
                    />
                    <div
                      class="react-select__control css-19jej77-control"
                    >
                      <div
                        class="react-select__value-container css-3tzzl5-ValueContainer"
                      >
                        <div
                          class="react-select__placeholder css-19jynah-placeholder"
                          id="react-select-7-placeholder"
                        >
                          Parent Sector
                        </div>
                        <div
                          class="react-select__input-container css-6j8wv5-Input"
                          data-value=""
                        >
                          <input
                            aria-autocomplete="list"
                            aria-describedby="react-select-7-placeholder"
                            aria-expanded="false"
                            aria-haspopup="true"
                            autocapitalize="none"
                            autocomplete="off"
                            autocorrect="off"
                            class="react-select__input"
                            id="react-select-7-input"
                            role="combobox"
                            spellcheck="false"
                            style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                            tabindex="0"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                      <div
                        class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                      >
                        <span
                          class="react-select__indicator-separator css-1979yuk-indicatorSeparator"
                        />
                        <div
                          aria-hidden="true"
                          class="react-select__indicator react-select__dropdown-indicator css-1hmmjq6-indicatorContainer"
                        >
                          <svg
                            class="text-2xl"
                            fill="none"
                            height="1em"
                            stroke="currentColor"
                            stroke-width="0"
                            viewBox="0 0 24 24"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="2"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                    <input
                      name="parentSector"
                      type="hidden"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="flex flex-col gap-y-2 mb-2"
                  data-testid="sector-displayable"
                >
                  <label
                    class="flex gap-x-2 mb-2 items-center self-start form-field-label"
                    for="sector-displayable"
                  >
                    <input
                      class="h-8 w-8"
                      data-testid="checkbox"
                      name="isDisplayable"
                      type="checkbox"
                    />
                    Is displayable
                  </label>
                </div>
              </div>
            </form>
          </div>
        </div>
        <div
          class="flex justify-start gap-3 text-xl"
        >
          <button
            class="submit-button"
            data-testid="submit-button"
            form="sector-form"
            type="submit"
          >
            Submit
          </button>
          <button
            class="cancel-button"
            data-testid="cancel-button"
            type="button"
          >
            Cancel
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
